### PR TITLE
changes to mutadone and mutagen

### PR DESCRIPTION
**Pull request by Tlaltecuhtli**
:cl:
balance: mutadone does 0.5 stamina damage, stops mutations from doing stuff when in the metabolism, once it depletes it ll remove 1 random mutation if it reached 50 cycles
balance: mutagen does 1 tox damage and it triggers mutations when in the blood stream instead of doing nothing
balance: the roundstart mutadone/mannitol pills have 10 u/20 u of reagents
/:cl:
mutadone:
the dna machine already can fix defects in a more interesting way, gives genetics something to do medical oriented other than giving hulk to assistants and tbh 0.1 u pill to cure any defects is heh

mutagen:
it did literally nothing while in the bloodstream other than fart level of radiations

### CODE REVIEW

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 1]
```diff
@@ -875,16 +875,37 @@
...
+	var/list/mut = list()
+
+/datum/reagent/medicine/mutadone/on_mob_metabolize(mob/living/carbon/M)
+	..()
+	var/len = M.dna.mutations.len
+	for(var/i in 1 to len)
+		mut += M.dna.mutations[i].type
 
 /datum/reagent/medicine/mutadone/on_mob_life(mob/living/carbon/M)
+	M.adjustStaminaLoss(0.5 *REM, 0)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: undoc'd?

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 2]
```diff
@@ -875,16 +875,37 @@
...
-	description = "Removes jitteriness and restores genetic defects."
+	description = "Removes jitteriness and temporarly supresses genes from manifesting, it also prevents new mutation from appearing while in the system."
 	color = "#5096C8"
 	taste_description = "acid"
+	overdose_threshold = 35
+	var/list/mut = list()
+
+/datum/reagent/medicine/mutadone/on_mob_metabolize(mob/living/carbon/M)
+	..()
+	var/len = M.dna.mutations.len
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: do all carbons have/use DNA?

**Comment by Repukan** [comment]
Wasn't there a chem that stopped mutations already

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 3]
```diff
@@ -875,16 +875,37 @@
...
 	if(M.has_dna())
 		M.dna.remove_all_mutations(mutadone = TRUE)
 	if(!QDELETED(M)) //We were a monkey, now a human
 		..()
+/datum/reagent/medicine/mutadone/on_mob_end_metabolize(mob/living/carbon/M)
+	..()
+	M.dna.remove_all_mutations(mutadone = TRUE) //just to be sure
+	var/len = mut.len
+	for(var/i in 1 to len)
+		M.dna.add_mutation(mut[i])
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: Is there a mutation that would reset progress from deleting/readding?

File: code/modules/reagents/reagent_containers/pill.dm [hunk 4]
```diff
@@ -133,14 +133,14 @@
 	name = "mannitol pill"
 	desc = "Used to treat brain damage."
 	icon_state = "pill17"
-	list_reagents = list(/datum/reagent/medicine/mannitol = 50)
+	list_reagents = list(/datum/reagent/medicine/mannitol = 20)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/reagent_containers/pill.dm]: what's with the random mannitol change?

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 5]
```diff
@@ -875,16 +875,37 @@
...
+	M.adjustStaminaLoss(0.5 *REM, 0)
 	M.jitteriness = 0
 	if(M.has_dna())
 		M.dna.remove_all_mutations(mutadone = TRUE)
 	if(!QDELETED(M)) //We were a monkey, now a human
 		..()
+/datum/reagent/medicine/mutadone/on_mob_end_metabolize(mob/living/carbon/M)
+	..()
+	M.dna.remove_all_mutations(mutadone = TRUE) //just to be sure
+	var/len = mut.len
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: remove

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 6]
```diff
@@ -875,16 +875,37 @@
...
+	for(var/i in 1 to len)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: mut.len

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 7]
```diff
@@ -875,16 +875,37 @@
...
+	description = "Removes jitteriness and temporarly supresses genes from manifesting, it also prevents new mutation from appearing while in the system."
 	color = "#5096C8"
 	taste_description = "acid"
+	overdose_threshold = 35
+	var/list/mut = list()
+
+/datum/reagent/medicine/mutadone/on_mob_metabolize(mob/living/carbon/M)
+	..()
+	var/len = M.dna.mutations.len
+	for(var/i in 1 to len)
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: M.dna.mutations.len

assuming all carbons have/use DNA. Pretty sure there's a check specifically for that.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 8]
```diff
@@ -875,16 +875,37 @@
...
+		mut += M.dna.mutations[i].type
 
 /datum/reagent/medicine/mutadone/on_mob_life(mob/living/carbon/M)
+	M.adjustStaminaLoss(0.5 *REM, 0)
 	M.jitteriness = 0
 	if(M.has_dna())
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: oh here it is

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 7 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: FYI you can also do

```DM
for (var/i in M.dna.mutations)
    var/datum/mutation/beep = i
    mut += beep.type
```

Just seems weird to convert to length then convert back to what the thing actually is when you can just loop through entries directly imo.

**Comment by Dawson1917** [comment]
Mutations gained from radiation storms and, I'm guessing, other random sources of radiation can't be turned off or copied in the genetics scanner. Mutadone is the only way to get rid of them. Also, having another player sit you in a scanner _and know how genetics even works_ every time you get mutated from something is rartedly unfun and not an interesting interaction in any way.

**Comment by Tlaltecuhtli** [comment]
rad storm does the same procs as mutagen which does work, are you sure about it?
if people dont know about x mechanic they ll just learn it like surgery, etc. you can always get 100 mutadone pills

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 3 above)
Comment (Tlaltecuhtli) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: what progress?

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 1 above)
Comment (Tlaltecuhtli) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: not sure if keep it, i wanted to make it feel like chemotherapy

Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: I don't care either way, just doc it.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: That's what i'm asking, is there any mutation that would reset some sort of progression mechanic by del/add?

Comment (Tlaltecuhtli) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: stuff like add timers?

**Comment by Krysonism** [comment]
I'd rather see mutadone's recipe being changed to require some chem from botany and the mechanic being changed to 10% remove a mutation per cycle to prevent cheesing with 0.0001u mutadone pills.

**Comment by Tlaltecuhtli** [comment]
i can just make it need a minimum of 5u to work

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm (hunk 3 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: not necessarily. Is there any perf-like mechanic where the mutation does something / changes in power over time?

I can't think of any atm so prob a nonissue.

**Comment by Krysonism** [comment]
I do like the mutagen change, if the mutadone changes turns out to be terrible after they are merged i will make a PR with my suggestion for mutadone.

**Comment by necromanceranne** [comment]
>buffs the shit out of mutagen so it just fucking ruins anyone hit with it
>even the smallest amount could mutate you so people are likely to griff with it even more
>makes mutadone useless
>without someone to help mutagen just permanently cripples you and essentially can remove you from the round

I love the direction we're going with this change very cool

**Comment by SpacePrius** [comment]
Why would you remove the only way to properly remove mutations besides fucking with them in the gene editor?

**Comment by SynnGraffkin** [comment]
4fded840c8168f2f25b0919b3ee300b1

Geneticists can barely handle doing cloning that is assigned to them and you expect them to cure people of mutations if Mutadone gets nerfed like that? Mutagen also does give mutations if you actually ingest it or put it in your meds and not "instead of doing nothing" as you put it. I've had to stop putting round start mutagen in the chem fridge because a certain clown would constantly come and chug most of it before the Botanist arrived, and normally that Clown would have all sorts of cancerous mutations. If you want mutations to be non-removable, there is already a system in game by using the Chromosomes.

**Comment by Tlaltecuhtli** [comment]
@Krysonism i dont mind it being made with botany help but then it would require some pills in the medvendor as there wont be type 1 availability

**Comment by necromanceranne** [comment]
Oh also undocumented changes AGAIN.

Why the fuck do you keep doing that? List your changes for crying out loud.

**Comment by Shadowflame909** [comment]
>Traitor Geneticist makes a shitload of radioactive/blind/monkey syringes.

>Decons the genetic machines.

Bro...why make an easy way to perma-fuck people.

File: code/modules/reagents/chemistry/reagents/medicine_reagents.dm [hunk 9]
```diff
@@ -875,17 +875,41 @@
 
 /datum/reagent/medicine/mutadone
 	name = "Mutadone"
-	description = "Removes jitteriness and restores genetic defects."
+	description = "Removes jitteriness and temporarly supresses genes from manifesting, it also prevents new mutation from appearing while in the system."
```
Comment (ATH1909) [inline code/modules/reagents/chemistry/reagents/medicine_reagents.dm]: new mutation_s_

**Comment by ATH1909** [comment]
https://www.youtube.com/watch?v=JaotiOp45dg

This PR is godawful. Not only does it increase the griefing power of unstable mutagen massively, but it also nerfs mutadone into the ground. With how you've done this, would you even be able to nullify someone's mutations while they have mutadone in their system, or would you have to wait for the mutadone to leave their system so that their mutations can re-manifest so that you can nullify them?

Does mutadone even fix genetic overload (y'know, that thing that happens when you get too much genetic instability) anymore with this change? It looks like it'd give you a temporary reprieve, then put you back on the 1 minute until horrific death timer again once it leaves your system.

also undocumented changes grrrrrrr

**Comment by ATH1909** [comment]
> I do like the mutagen change, if the mutadone changes turns out to be terrible after they are merged i will make a PR with my suggestion for mutadone.

"after they are merged"

NO

BAD COBBY

Even a day of this PR being merged would be hell (once the clown, mime, botanist, chemist, and/or geneticist mains find out about it). Instead of merging a PR that you know is bad and then fixing it a day or two later, why not just not merge that horrible PR in the first place and release a fixed version of the PR later? In other words, close this PR and open a new one that's just, say, the unstable mutagen change (which is still bad, but not as AWFUL as a PR with both the proposed mutadone nerf and the proposed unstable mutagen buff).

**Comment by mrhugo13** [comment]
I was gonna make some long rant about this but i literally only need to say this:
Chemist griefing would go up by tenfold.

**Comment by necromanceranne** [comment]
The mutagen change would be somewhat okay on it's own, since it gives mutagen some bite. But the double whammy is just too much to me.

**Comment by 81Denton** [comment]
Isn't this a pretty big buff to hulks when you can no longer dehulk them permanently with mutadone

**Comment by Shadowflame909** [comment]
Hulks already got made useless when everyone realized disablers stunned them.

Who cares about those nerds

**Comment by OmegaShoots** [comment]
Mutagen causes a mutation on ingestion, and damages you while metabolizing.
That's certainly more than "nothing" and allows risky chemists/maint meth lab assistants to roll the dice on getting a mutation.

**Comment by Wilchenx** [comment]
>Chemistry buffs

unstable mutagen is already stupid easy to make, if you're buffing it atleast make the recipe way more pricy

15u syringes of mutagen will absolutely destroy people with random mutations

**Comment by 81Denton** [comment]
This is too big of a balance change at once and would allow people to cripple other players with mutagen way too easy. Regular toxins can at least be purged and the victim defibbed.

**Comment by ExcessiveUseOfCobblestone** [comment]
I'd much rather have mutadone be a somewhat temporary solution to genetic defects as opposed to going to genetics itself.

It would make more sense for genetics to fix genetic disabilities rather than chemistry anyways.

> > Traitor Geneticist makes a shitload of radioactive/blind/monkey syringes.
> 
> > Decons the genetic machines.
> 
> Bro...why make an easy way to perma-fuck people.

The endgoal would be for someone to make mutadone until you can get a new scanner going. Sabotage is a part of the game, not a design flaw.

**Comment by Shadowflame909** [comment]
Genetics has so many ways to ruin your round with major defects though. Generally, the coders who made these defects knew there was an easy way to remove them.

If you take that out, it's just another unfun way to inflict mental pain upon others. Imagine a geneticist stunning you roundstart and injecting you with radioactivity. It could spiral into a quite literally round ender because of the way radiation works.

The feature just wasn't intended to be permanent, and I don't think anyones going to majorly change genetics to where it being permanent doesn't matter.

It'll screw players, about as much skewium and shoe cuffs do but more permanent. Because of how easy it is to make, it's just an easier form of those two i say.

**Comment by ExcessiveUseOfCobblestone** [comment]
where are they getting this roundstart radioactivity? Traitor chem kit?

If the issue is that these things were "balanced around (mutadone existing)" then they clearly weren't balanced well because that supply can easily be removed. The theoretical "what if everything goes wrong and your round is made not fun?!!!!!!" doesn't really move me.

Yes, if a geneticist manages to lolmeme u with enough rads that give you crippling disability spam and then he sabotages the scanner and then he sabotages all forms of retrieving a new scanner and then he demolishes chemistry so you can't even get temp mutadone all while no one has prompted to kill him yet then yes, you're screwed. Unfortunate for you.

**Comment by Shadowflame909** [comment]
It's not that hard to get a negative trait at roundstart is what I'm saying. This is intentional since you should be scouring for the rare good ones. You're suggesting that this job should be your main source of relief against its own punishment.

It's like having an arrested criminal write the laws. He's not going to stop himself. Ultimately, it'd just let genetics get away with a whole lot more traitoring. If the geneticist is bad. Then the geneticist won't stop himself when he whips out the Unintelligible or Spacial Instability that he easily got roundstart

It's not theoretical. It's pointless. You've futilely designed a system where the only one who can stop crimes is the one who's able to do so in the first place. It'll make rounds end faster, and it's a botched idea that pretty much goes against any goals the maintainers have strived for.

Having two permanent ways to fix things is always an option for everything else. That shouldn't change.

**Comment by ExcessiveUseOfCobblestone** [comment]
Like Chem treating Toxins or Viro Treating Bad Diseases???

**Comment by Shadowflame909** [comment]
>The only one who can stop

Doesn't apply to either of those unlike here. Since you need a special machine.

**Comment by ExcessiveUseOfCobblestone** [comment]
You mean like a Pandemic or a Chem dispenser?

**Comment by Shadowflame909** [comment]
Gonna use one of your own arguments cobblestone. 

It's different now thanks to a special feature calling the medical vendor.  There's also the holodeck. Still, ignoring the fact virology is self-neutering when it comes to making deadly things. Doesn't really go against the point that this method of buffing antagonists will thoroughly do its job of lowering round-times.

**Comment by ExcessiveUseOfCobblestone** [comment]
Yes you're right in that the roundstart supplies can mitigate toxins, but that doesn't address viro.

You could say spaceacillin I guess but that would be a mitigator rather than a cure.

Holodeck medical is up for removal, I CBA to look at the justification for why it was added to begin with.

I don't consider buffing antagonists this way any different than just letting them run wild with a sword and ebow then spacing/permaremoving. I doubt very seriously the round time will take a note-worthy hit with the change. We've had an allegedly huge round-time shortener with the sleeper addition and I don't see a difference, and that was a much more noteworthy "buff to antags".

**Comment by Shadowflame909** [comment]
>Holodeck medical is up for removal
Youch.

**Pull request closed without merging**